fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -156,31 +157,31 @@ public void commit() {
                 transaction.mutate(valuePair.getLeft(), key, valuePair.getRight()));
         CompletableFuture<byte[]> versionFuture = transaction.getVersionstamp();
         long beforeCommitTimeMillis = System.currentTimeMillis();
-        CompletableFuture<Void> commit = checks.isDone() && !checks.isCompletedExceptionally() ?
+        CompletableFuture<Void> commit = MoreAsyncUtil.isCompletedNormally(checks) ?
```

Comment:
I suspect that this will conflict with #464. Or maybe they won't because they make the same change?